src/coreclr/src/jit/optimizer.cpp

```diff
@@ -8659,6 +8659,145 @@ bool Compiler::optIsRangeCheckRemovable(GenTree* tree)
     return true;
 }
 
+//----------------------------------------------------------------------------------
+// optBranchlessConditions: optimize simple expressions to branchless expressions
+//
+//  Operation:
+//      Look for BBJ_COND basic blocks which target two BBJ_RETURN blocks
+//      and convert them into single branchless expressions, e.g.
+//
+//            JTRUE
+//              \
+//              EQ              RETURN
+//             /  \                \
+//            X    Y               ADD
+//                       =>       /   \
+//           RETURN             EQ     C3
+//              \              /  \
+//              C1            X    Y
+//
+//           RETURN
+//              \
+//               C2
+//
+void Compiler::optBranchlessConditions()
+{
+    JITDUMP("\n*************** In optBranchlessConditions()\n");
+    bool changed = false;
+    for (BasicBlock* block = fgFirstBB; block; block = block->bbNext)
+    {
+        if ((block->bbJumpKind != BBJ_COND) || block->isEmpty())
+        {
+            continue;
+        }
+
+        Statement* fistStmt = block->lastStmt();
+        if ((fistStmt == nullptr) || !fistStmt->GetRootNode()->OperIs(GT_JTRUE))
+        {
+            continue;
+        }
+
+        GenTree*  rootNode    = fistStmt->GetRootNode();
+        GenTree*  rootSubNode = rootNode->gtGetOp1();
+        var_types typ         = rootSubNode->TypeGet();
+
+        assert(rootSubNode != nullptr);
+
+        if (!varTypeIsIntegral(typ) || (rootNode->gtFlags & GTF_ORDER_SIDEEFF))
+        {
+            continue;
+        }
+
+        BasicBlock* trueBb  = block->bbJumpDest;
+        BasicBlock* falseBb = block->bbNext;
+
+        assert((trueBb != nullptr) && (falseBb != nullptr));
+
+        if (!rootSubNode->OperIsCompare() || (trueBb == falseBb) ||
+            // both blocks are expected to be BBJ_RETURN
+            // TODO: implement for GT_ASG
+            (trueBb->bbJumpKind != BBJ_RETURN) || (falseBb->bbJumpKind != BBJ_RETURN) ||
+            // give up if these blocks are used by someone else
+            (trueBb->countOfInEdges() != 1) || (falseBb->countOfInEdges() != 1) ||
+            (trueBb->bbFlags & BBF_DONT_REMOVE) || (falseBb->bbFlags & BBF_DONT_REMOVE))
+        {
+            continue;
+        }
+
+        Statement* trueBbStmt  = trueBb->firstStmt();
+        Statement* falseBbStmt = falseBb->firstStmt();
+        if ((trueBbStmt == nullptr) || (trueBbStmt->GetPrevStmt() != trueBbStmt) ||
+            // make sure both blocks are single statement
```

Comment:
> // make sure both blocks are single statement [](start = 11, length = 46)

nit: have the comment before if-condition instead of middle of it.

Reply:
I just didn't want clang-format to make it less readable so I separated groups of conditions with comments (I remove the comments - clang-foramat will force me to re-format it)

Reply:
got it.